specutils/io/default_loaders/jwst_reader.py

```diff
@@ -0,0 +1,65 @@
+import astropy.units as u
+from astropy.io import fits
+
+from ...spectra import Spectrum1D, SpectrumList
+from ..registers import data_loader
+
+
+def identify_jwst_fits(origin, *args, **kwargs):
+    """
+    Check whether the given file is a JWST spectral data product.
+
+    This check is fairly simple. It expects FITS files that contain an ASDF
+    header (which is not used here, but indicates a JWST data product). It then
+    looks for at least one EXTRACT1D header, which contains spectral data.
```

Comment:
I know I'm being a bit paranoid/overly future-proofing here, so you can feel free to ignore this if there's not an obvious answer... but perhaps it should actually look for some processing-specific metadata?  Like something that actually says "JWST" somewhere, or a list of instruments or something?

That is, suppose some of the JWST pipelines are re-used by some other mission. They might end up with `EXTRACT1D`, and it's counter-intuitive that the JWST reader would work on them.